app/controllers/sessions_controller.rb

```diff
@@ -49,6 +58,12 @@ def ask_new_or_returning
   def i_am_returning
   end
 
+  # This is an official action instead of just doing `redirect_back` in callback
+  # handler so that fine_print can check to see if terms need to be signed.
```

Comment:
I think this is unnecessary? FinePrint should check that on the doorkeeper authorize route.